include/openssl/lhash.h

```diff
@@ -86,9 +86,17 @@ void OPENSSL_LH_doall(OPENSSL_LHASH *lh, OPENSSL_LH_DOALL_FUNC func);
 void OPENSSL_LH_doall_arg(OPENSSL_LHASH *lh, OPENSSL_LH_DOALL_FUNCARG func, void *arg);
 unsigned long OPENSSL_LH_strhash(const char *c);
 unsigned long OPENSSL_LH_num_items(const OPENSSL_LHASH *lh);
+
+/*
+ * We would like to deprecate these functions, but doing so gets errors in
+ * builds when allowing deprecated, and we'd have to annotate this file
+ * which is include'd in several other files, so that's not good.
+ * Instead, we just declare these as normal, and the |DEFINE_LASH_OF|
+ * macro includes the type-safe wrapper functions or not, depending on
+ * deprecation.
+ */
 unsigned long OPENSSL_LH_get_down_load(const OPENSSL_LHASH *lh);
 void OPENSSL_LH_set_down_load(OPENSSL_LHASH *lh, unsigned long down_load);
-
 # ifndef OPENSSL_NO_STDIO
 void OPENSSL_LH_stats(const OPENSSL_LHASH *lh, FILE *fp);
```

Comment:
These should be wrapped by the `DEPRECATEDIN_3_0` macro to give users a warning that they are going.
This would then change `util/libcrypto.num`.

Or have I missed something subtle about using them subsequently in the type-safe wrappers?

Reply:
Nobody should be using those functions directly, and the type-safe macro marks them correctly (see ```LHASH_OF_TYPE_DEPRECATED```).  Is that sufficient?

Reply:
I do not see the DEPRECATEDIN_3_0 in the type safe macros either.